main_server/local_server/server_admin/server_admin_javascript.js

```diff
@@ -1396,9 +1396,29 @@ function BMLT_Server_Admin()
     /************************************************************************************//**
     *   \brief
     ****************************************************************************************/
+    this.checkURL = function(string) {
+        let url;
+        try {
+           url = new URL(string);
+        } catch (_) {
+           return false;
```

Comment:
I love appropriate usages of exceptions as control flow... nice.

Old article: https://web.archive.org/web/20140430044213/http://c2.com/cgi-bin/wiki?DontUseExceptionsForFlowControl